crates/neon-runtime/src/napi/tsfn.rs

```diff
@@ -167,6 +179,12 @@ impl<T: Send + 'static> ThreadsafeFunction<T> {
 
 impl<T> Drop for ThreadsafeFunction<T> {
     fn drop(&mut self) {
+        // tsfn was already finalized by `Environment::CleanupHandles()` in
+        // Node.js
+        if *self.is_finalized.lock().unwrap() {
```

Comment:
I'm a bit worried about the cases where we might end up running `release` concurrently to `finalize` callback. Maybe this has to be fixed in Node.js instead...